rust/extns/numaflow-sqs/src/source.rs

```diff
@@ -0,0 +1,751 @@
+        Self {
+            handler_rx,
+            client,
+            queue_url,
+        }
+    }
+
+    async fn create_sqs_client(config: Option<SqsSourceConfig>) -> Client {
+        let region = match config {
+            Some(config) => config.region.clone(),
+            None => SQS_DEFAULT_REGION.to_string(),
+        };
+
+        tracing::info!(region = region.clone(), "Creating SQS client in region");
+
+        // read aws config
+        let region_provider = RegionProviderChain::first_try(Region::new(region.clone()))
+            .or_default_provider()
+            .or_else(Region::new(SQS_DEFAULT_REGION));
+
+        let shared_config = aws_config::defaults(BehaviorVersion::v2024_03_28())
+            .region(region_provider)
+            .load()
+            .await;
+
+        // create sqs client
+        Client::new(&shared_config)
+    }
+
+    async fn run(&mut self) {
+        while let Some(msg) = self.handler_rx.recv().await {
+            self.handle_message(msg).await;
+        }
+    }
+
+    async fn handle_message(&mut self, msg: SqsActorMessage) {
+        match msg {
+            SqsActorMessage::Receive {
+                respond_to,
+                count,
+                timeout_at,
+            } => {
+                let messages = self.get_messages(count, timeout_at).await;
+                respond_to
+                    .send(messages)
+                    .expect("failed to send response from SqsActorMessage::Receive");
+            }
+            SqsActorMessage::Delete {
+                respond_to,
+                offsets,
+            } => {
+                let status = self.delete_messages(offsets).await;
+                respond_to
+                    .send(status)
+                    .expect("failed to send response from SqsActorMessage::Delete");
+            }
+            SqsActorMessage::GetPending { respond_to } => {
+                let status = self.get_pending_messages().await;
+                respond_to
+                    .send(status)
+                    .expect("failed to send response from SqsActorMessage::GetPending");
+            }
+        }
+    }
+
+    /// Retrieves messages from SQS with timeout and batching.
+    ///
+    /// Implementation details:
+    /// - Handles queue URL caching/refresh
+    /// - Respects timeout for long polling
+    /// - Processes message attributes and system metadata
+    /// - Returns messages in a normalized format
+    async fn get_messages(&mut self, count: i32, timeout_at: Instant) -> Result<Vec<SqsMessage>> {
+        let remaining_time = timeout_at - Instant::now();
+
+        let sdk_response = self
+            .client
+            .receive_message()
+            .queue_url(self.queue_url.clone())
+            .max_number_of_messages(count)
+            .message_attribute_names("All")
+            .message_system_attribute_names(MessageSystemAttributeName::All)
+            .wait_time_seconds(remaining_time.as_secs() as i32)
```

Comment:
Support at milliseconds, users do set read timeout in millis

Reply:
aws sqs sdk does not support milliseconds, is there any other way to support milliseconds?